### ISSUE TITLE
model register_buffer missing persistent

### ISSUE THREAD

**Issue by williamlzw** [dotnet/TorchSharp#1141]
//python   
       class CLIPTextEmbeddings(nn.Module):
            def __init__(self, n_vocab: int, n_embd: int, max_position_embeddings: int):
                super().__init__()
                self.token_embedding = nn.Embedding(n_vocab, n_embd)
                self.position_embedding = nn.Embedding(max_position_embeddings, n_embd)
                self.register_buffer("position_ids", torch.arange(max_position_embeddings).expand((1, -1)), persistent=False)
        
            def forward(self, input_ids):
                seq_length = input_ids.shape[-1]
                position_ids = self.position_ids[:,:seq_length]
                inputs_embeds = self.token_embedding(input_ids)
                position_embeddings = self.position_embedding(position_ids)
                embeddings = inputs_embeds + position_embeddings
                return embeddings
    
    //c#
    register_buffer("position_ids", torch.arange(max_position_embeddings).expand(new long[] { 1, -1 }), persistent = false);//error

**Comment by NiklasGustafsson** [comment]
That should be easy enough to fix.

**Comment by NiklasGustafsson** [comment]
In Python, a non-persistent buffer will be excluded by the state_dict(), but is then usable as an attribute (field, if you will) from Python code, as the example above shows -- `self.position_ids`.

What would be the corresponding C# usage -- you would have to declare it as a field to have access to it in this way, but you want it to be excluded from consideration in the state_dict, is that right?

**Comment by williamlzw** [comment]
There's an alternate way to write it.
            public class CLIPTextEmbeddings : Module<Tensor, Tensor>
            {
                private readonly Embedding token_embedding;
                private readonly Embedding position_embedding;
                private readonly long max_position_embedding;
            
                public CLIPTextEmbeddings(long n_vocab, long n_embd, long max_position_embeddings) : base("CLIPTextEmbeddings")
                {
                    token_embedding = Embedding(n_vocab, n_embd);
                    position_embedding = Embedding(max_position_embeddings, n_embd);
                    max_position_embedding = max_position_embeddings;
                    RegisterComponents();
                }
            
                public override Tensor forward(Tensor input_ids)
                {
                    using var _ = NewDisposeScope();
                    var seq_length = input_ids.shape[input_ids.dim() - 1];
                    var position_ids = torch.arange(max_position_embedding).expand(new long[] { 1, -1 });
                    var position_id = position_ids[torch.TensorIndex.Colon, torch.TensorIndex.Slice(0, seq_length)];
                    var inputs_embeds = token_embedding.forward(input_ids);
                    var position_embeddings = position_embedding.forward(position_id);
                    var embeddings = inputs_embeds + position_embeddings;
                    return embeddings.MoveToOuterDisposeScope();
                }
            }

**Comment by NiklasGustafsson** [comment]
Yes, you can always create a local variable, but if you want to save some time in `forward()` the non-persistent buffer makes sense.
